sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/constraints.scala

```diff
@@ -112,6 +117,27 @@ case class CheckConstraint(
   with TableConstraint {
 // scalastyle:on line.size.limit
 
+  def toV2Constraint(isCreateTable: Boolean): Constraint = {
+    val predicate = new V2ExpressionBuilder(child, true).buildPredicate().orNull
+    val enforced = userProvidedCharacteristic.enforced.getOrElse(true)
+    val rely = userProvidedCharacteristic.rely.getOrElse(false)
+    // The validation status is set to UNVALIDATED for create table and
+    // VALID for alter table.
+    val validateStatus = if (isCreateTable) {
+      Constraint.ValidationStatus.UNVALIDATED
+    } else {
+      Constraint.ValidationStatus.VALID
```

Comment:
Is the idea here that we always validate existing data in ALTER?

Reply:
Yes for check constraint